### PR TITLE
Add catalog hash to catalog_counters xattr

### PR DESCRIPTION
**Pull request by HereThereBeDragons**
catalog hash is returned for file when calling `attr -g catalog_counters <file>`

### CODE REVIEW

**Comment by jblomer** [comment]
Fixes #2900

File: cvmfs/catalog_mgr.h [hunk 1]
```diff
@@ -201,7 +201,8 @@ class AbstractCatalogManager : public SingleCopy {
   }
 
   catalog::Counters LookupCounters(const PathString &path,
-                                   std::string *subcatalog_path);
+                                   std::string *subcatalog_path,
+                                   shash::Any* hash);
```
Comment (jblomer) [inline cvmfs/catalog_mgr.h]: Style nit.
```suggestion
                                   shash::Any *hash);
```

File: cvmfs/catalog_mgr_impl.h [hunk 2]
```diff
@@ -653,6 +655,7 @@ catalog::Counters AbstractCatalogManager<CatalogT>::LookupCounters(
     }
   }
 
+  *hash = catalog->hash();
```
Comment (jblomer) [inline cvmfs/catalog_mgr_impl.h:659]: Let's set the hash to `shash::Any()` in the error path above (L651)

File: test/src/701-xattr-catalog_counters/setup_teardown [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+teardown() {
+  echo "*** TEARDOWN of repository named $CVMFS_TEST_REPO of user $CVMFS_TEST_USER"
+
+  ## ERROR: THIS COMMAND DOES NOT WORK
+	# destroy_repo
+
+  sudo cvmfs_server rmfs -f $CVMFS_TEST_REPO
+  echo "*** FINISHED teardown of repository named $CVMFS_TEST_REPO of user $CVMFS_TEST_USER"
+}
```
Comment (jblomer) [inline test/src/701-xattr-catalog_counters/setup_teardown]: missing newline

File: test/src/701-xattr-catalog_counters/setup_teardown [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  touch /cvmfs/$CVMFS_TEST_REPO/foo/foo1.txt
+  touch /cvmfs/$CVMFS_TEST_REPO/foo/foo2.txt
+  touch /cvmfs/$CVMFS_TEST_REPO/foo/bar/.cvmfscatalog
+  touch /cvmfs/$CVMFS_TEST_REPO/foo/bar/bar1.txt
+  touch /cvmfs/$CVMFS_TEST_REPO/foo/bar/bar2.txt
+  publish_repo $CVMFS_TEST_REPO || return 2
+  echo "*** FINISHED creating a fresh repository named $CVMFS_TEST_REPO with user $CVMFS_TEST_USER"
+}
+
+teardown() {
```
Comment (jblomer) [inline test/src/701-xattr-catalog_counters/setup_teardown]: We typically leave the test repository up and running and let the next test destroy it. That could be changed but then it should be done consistently, I think.

File: test/src/701-xattr-catalog_counters/main [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+  echo "*** FINISHED check catalog hashes"
+}
+
+cvmfs_run_test() {
+  logfile=$1
+
+  create_nested_repo
+
+  check_catalog_hashes
```
Comment (jblomer) [inline test/src/701-xattr-catalog_counters/main]:
```suggestion
  check_catalog_hashes || return $?
```

File: test/src/701-xattr-catalog_counters/main [hunk 6]
```diff
@@ -0,0 +1,61 @@
+source ./src/701-xattr-catalog_counters/setup_teardown
```
Comment (jblomer) [inline test/src/701-xattr-catalog_counters/main]: Let's source that after the preamble of paramters

File: test/src/701-xattr-catalog_counters/main [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+  local hash_root=$(echo "$list_catalogs" | grep '/$' | awk -F' ' '{print $1}')
+  local hash_foo=$(echo "$list_catalogs" | grep '/foo$' | awk -F' ' '{print $1}')
+  local hash_bar=$(echo "$list_catalogs" | grep '/bar$' | awk -F' ' '{print $1}')
+ 
+  echo "Retrieve catalog hashes for files"
+  local catalog_hash_root1=$(attr -g catalog_counters /cvmfs/$CVMFS_TEST_REPO/root1.txt | grep 'hash' | awk -F' ' '{print $2}' ) 
+  local catalog_hash_root2=$(attr -g catalog_counters /cvmfs/$CVMFS_TEST_REPO/root2.txt | grep 'hash' | awk -F' ' '{print $2}' ) 
+
+  ## THIS IS NOT POSSIBLE BECAUSE ROOT DIR
+  # local catalog_hash_root_dir=$(attr -g catalog_counters /cvmfs/$CVMFS_TEST_REPO/ | grep 'hash' | awk -F' ' '{print $2}' )
```
Comment (jblomer) [inline test/src/701-xattr-catalog_counters/main]: This should be possible... do you get an error?

**Review by jblomer — CHANGES_REQUESTED** [review]
Very nice!  Just a few minor comments (and the linter complaints).

File: test/src/701-xattr-catalog_counters/main (hunk 7 above)
Comment (HereThereBeDragons) [reply test/src/701-xattr-catalog_counters/main]: it is possible but the result is empty. with `grep` and `awk` behind it they start complaining about to work on an empty string

**Pull request closed without merging**